### PR TITLE
Add resource suggestion

### PR DESCRIPTION
**Pull request by lemonlatte**
In this PR, we add new API for suggesting resources for user. It first collection distinct resources that have been rated by a user. Then, it unions those resources with pre-defined important resources and returns them.

Changes Highlight
- Extend the `getResourceList` to make it be able to return important resources directly
- Create an aggregation module to gather used aggregations so we can re-use it later

### CODE REVIEW

File: api/suggestion.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		return
+	}
+
+	lang := "en"
+	if params.Language != "" {
+		lang = params.Language
+	}
+
+	if !params.Suggestion {
+		abortWithEncoding(c, http.StatusInternalServerError, errorResourceNotSupport)
```
Comment (nodestory) [inline api/suggestion.go:35]: What's `params.Suggestion` for?

Comment (lemonlatte) [reply api/suggestion.go:35]: It is to make this API work like this: `/api/resources?lang=zh&suggestion=true`. We do not have rest resource for `resource` now.

File: api/errorcode.go [hunk 2]
```diff
@@ -12,6 +12,7 @@ var (
 
 		1006: "invalid value of client version",
 		1007: "API for this client version has been discontinued",
+		1008: "resource is not support",
```
Comment (nodestory) [inline api/errorcode.go:15]:
```suggestion
		1008: "resource is not supported",
```

File: store/suggestion_test.go [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+	s.NoError(err)
+	s.Len(list, 0)
+}
+
+func (s *SuggestionTestSuite) TestListSuggestedResourceForAccountNotFound() {
+	store := NewMongoStore(s.mongoClient, s.testDBName)
+
+	list, err := store.ListSuggestedResources("test-account-not-found", "en")
+	s.NoError(err)
+	s.Len(list, 0)
```
Comment (nodestory) [inline store/suggestion_test.go:216]: I thought the user will still get 30 important resources.

File: store/suggestion_test.go [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+	s.NoError(err)
+	s.Len(list, 32)
+}
+
+func (s *SuggestionTestSuite) TestListSuggestedResourceForProfileWihtoutPOI() {
+	store := NewMongoStore(s.mongoClient, s.testDBName)
+
+	list, err := store.ListSuggestedResources("test-account-profile-id-no-poi", "en")
+	s.NoError(err)
+	s.Len(list, 0)
```
Comment (nodestory) [inline store/suggestion_test.go:208]: I thought the user will still get 30 important resources.

Comment (lemonlatte) [reply store/suggestion_test.go]: Fixed.

**Pull request merged**